core/src/main/java/org/eclipse/hono/util/ResourceLimits.java

```diff
@@ -55,13 +57,35 @@ public final ResourceLimits setMaxConnections(final int maxConnections) {
     /**
      * Gets the maximum number of connected devices a tenant supports.
      * 
-     * @return The maximum number of connections or {@link TenantConstants#DEFAULT_MAX_CONNECTIONS}
+     * @return The maximum number of connections or {@link TenantConstants#UNLIMITED_CONNECTIONS}
      *         if not set.
      */
     public final int getMaxConnections() {
         return this.maxConnections;
     }
 
+    /**
+     * Sets the maximum time-to-live to use for events published by
+     * devices of a tenant.
+     * 
+     * @param maxTtl The time-to-live in seconds.
+     * @return A reference to this for fluent use.
+     */
+    public ResourceLimits setMaxTtl(final long maxTtl) {
```

Comment:
It would be good to check whether `maxTtl <-1` here and throw an IllegalArgumentException if that's the case.
Otherwise there would be an exception in the proton code if `maxTtl < -1` (in Message.setTtl()).